src/sentry/digests/notifications.py

```diff
@@ -152,7 +168,9 @@ def rewrite_record(record, project, groups, rules):
     )
 
 
-def group_records(groups, record):
+def group_records(
+    groups: MutableMapping[str, Mapping[str, List[Record]]], record: Record
```

Comment:
`Sequence` instead of `List`?

Reply:
Unfortunately, `Sequence` doesn't have `.append()` so the param needs to use 'List`.

Reply:
I think you need `MutableSequence`